tests/test_cli_compile.py

```diff
@@ -3562,3 +3562,24 @@ def test_origin_of_extra_requirement_not_written_to_annotations(
         )
         == out.stdout
     )
+
+
+@pytest.mark.xfail(reason="https://github.com/jazzband/pip-tools/issues/2012")
+@mock.patch("piptools.scripts.compile.parse_requirements")
+def test_stdout_should_not_be_read_when_stdin_is_not_a_plain_file(
+    parse_req,
+    runner,
+    tmp_path,
+):
+    parse_req.side_effect = pytest.fail("Must not be called when output is a fifo")
```

Comment:
This should probably be wrapped into a lambda.

Reply:
Indeed yes.